src/phases/query/CollScanPredicateSelectivity.yml

```diff
@@ -0,0 +1,427 @@
+            quarterNull: {^Choose: {from: ["non-null", null], weights: [75, 25]}},
+            onePercentPositiveInt: {^Choose: {
+              from: &PositiveAndNegativeIntegers [{^Inc: {start: 1, step: 1}}, {^Inc: {start: -1, step: -1}}], 
+              weights: [1, 99]
+            }},
+            tenPercentPositiveInt: {^Choose: {from: *PositiveAndNegativeIntegers, weights: [10, 90]}},
+            quarterPositiveInt: {^Choose: {from: *PositiveAndNegativeIntegers, weights: [25, 75]}},
+            halfPositiveInt: {^Choose: {from: *PositiveAndNegativeIntegers, weights: [50, 50]}},
+            quarterNegativeInt: {^Choose: {from: *PositiveAndNegativeIntegers, weights: [75, 25]}},
+            tenPercentNegativeInt: {^Choose: {from: *PositiveAndNegativeIntegers, weights: [90, 10]}},
+            onePercentNegativeInt: {^Choose: {from: *PositiveAndNegativeIntegers, weights: [99, 1]}},
+            nested: [
+              {
+                onePercentNull: {^Choose: {from: ["non-null", null], weights: [99, 1]}},
+                tenPercentNull: {^Choose: {from: ["non-null", null], weights: [90, 10]}},
+                quarterNull: {^Choose: {from: ["non-null", null], weights: [75, 25]}},
+                onePercentPositiveInt: {^Choose: {from: *PositiveAndNegativeIntegers, weights: [1, 99]}},
+                tenPercentPositiveInt: {^Choose: {from: *PositiveAndNegativeIntegers, weights: [10, 90]}},
+                quarterPositiveInt: {^Choose: {from: *PositiveAndNegativeIntegers, weights: [25, 75]}},
+                halfPositiveInt: {^Choose: {from: *PositiveAndNegativeIntegers, weights: [50, 50]}},
+                quarterNegativeInt: {^Choose: {from: *PositiveAndNegativeIntegers, weights: [75, 25]}},
+                tenPercentNegativeInt: {^Choose: {from: *PositiveAndNegativeIntegers, weights: [90, 10]}},
+                onePercentNegativeInt: {^Choose: {from: *PositiveAndNegativeIntegers, weights: [99, 1]}},
+              }
+            ]
+          }
+
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [2]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+
+# In actors' names, a 'good case' means that the ordering of predicates defined by actual
+# selectivities matches the ordering defined by heuristic selectivity estimation. Heuristic CE 
+# defines the following selectivity order for predicates:
+# $eq < closed range (e.g. {$gt: 10, $lte: 20}) < open range (e.g. {$gt: 10}) < $ne
+
+- ActorFromTemplate:
+    TemplateName: FindQueryTemplate
+    TemplateParameters:
+      Name: HighSelHeuristicGoodTwoClauses
+      ActivePhase: 3
+      Filter:
+        # Here (0, 1e9) is the most selective predicate, both in reality and according to heuristic CE.
+        # Note that without predicate reordering, predicates are ordered by first comparing field names.
```

Comment:
for my own benefit does 'predicates are ordered by first comparing field names' mean that that onePercentNegativeInt would be applied first?